### ISSUE TITLE
Deduplicate `structuredClone()`

### ISSUE THREAD

**Issue by andreubotella** [denoland/deno#12655]
Although the implementation of `self.structuredClone()` that is exposed in the global object is the one at `ext/web/13_message_port.js`, there is one other implementation at `ext/web/02_structured_clone.js`. The former was added by @satyarohith in #10818 because it was needed for implementing the user timing spec, before @crowlKats implemented the latter in #11572.

The implementation at `ext/web/02_structured_clone.js` is not a spec-compliant implementation, though. It has performance optimizations that try to avoid cloning `ArrayBuffer`s, but they throw type errors at runtime – and even if they didn't, those performance optimizations would be observable from the user timing API.

It seems like in the long run it would be best to use `02_structured_clone.js` for `self.structuredClone()` and related internal functions – `(de)serializeJsMessageData` and perhaps the contents of #12505's `ext/web/00_interfaces.js`. This is not possible to do right now though, because `deserializeJsMessageData` must be able to create `MessagePort` objects. Once #12067 is further along and transfer receiving can be implemented using a `Deno.core` API, this would no longer be an issue.

**Comment by bartlomieju** [comment]
This issue appears to be outdated at this point. Both files still exist, but the architecture has evolved significantly since 2021:

- The global `structuredClone` is wired to the `13_message_port.js` version (full spec-compliant implementation with transfer support)
- Both implementations now delegate to `core.structuredClone()` for the actual cloning work, so the duplication is minimal
- `02_structured_clone.js` serves as a lightweight internal utility used only by `PerformanceEntry` and streams — it's not exposed to users, so the original spec-compliance concern is less relevant
- The recent cloneable resource registry work (#32672) shows the architecture has continued evolving with core handling the heavy lifting

The remaining separation is arguably intentional (fast-path internal utility vs full public API with WebIDL validation and transfer support). Closing as outdated.

**Comment by bellahadid27211-lab** [comment]
Yes it outdated they need to find a way to close issues that like this one
